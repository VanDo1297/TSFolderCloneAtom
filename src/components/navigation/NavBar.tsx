import React, {Component} from 'react';
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";
import {RouteComponentProps, withRouter} from "react-router";
import withSystemState, {InjectedSystemStateProps} from "../hocs/WithSystemState";

type NavBarProps = RouteComponentProps & InjectedSystemStateProps;

class NavBar extends Component<NavBarProps,any> {
    logOut = () =>{
        this.props.removeToken();
        this.props.history.push("/");
    };

    goTo = (destination:string) =>{
        this.props.history.push(destination);
    };

    render() {
        return(
            <Navbar bg="dark" variant="dark">
                <Navbar.Brand>Bohe</Navbar.Brand>
                <Nav className="mr-auto">
                    <Nav.Link onClick={()=>this.goTo("/home")}>Home</Nav.Link>
                    <Nav.Link onClick={()=>this.goTo("/cards")}>Cards</Nav.Link>
                    <Nav.Link onClick={()=>this.goTo("/charges")}>Charges</Nav.Link>
                    <Nav.Link onClick={()=>this.goTo("/subscriptions")}>Subscriptions</Nav.Link>
                    <Nav.Link onClick={()=>this.goTo("/invoices")}>Invoices</Nav.Link>
                    <Nav.Link onClick={this.logOut}>Log Out</Nav.Link>
                </Nav>
            </Navbar>
        );
    }
}

export default withSystemState(withRouter(NavBar));