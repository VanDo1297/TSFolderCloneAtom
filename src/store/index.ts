import {
  systemReducer,
  registerReducer,
  accountReducer,
  productReducer,
  profileReducer,
} from "./system/reducers";
import { combineReducers } from "redux";

export const rootReducer = combineReducers({
  system: systemReducer,
  register: registerReducer,
  account: accountReducer,
  product: productReducer,
  profile: profileReducer,
});

export type AppState = ReturnType<typeof rootReducer>;
