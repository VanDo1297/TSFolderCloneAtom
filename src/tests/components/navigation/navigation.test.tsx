import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from "../../../components/navigation/NavBar";
import {createStore} from "redux";
import {rootReducer} from "../../../store";
import {Provider} from "react-redux";
import {BrowserRouter as Router} from "react-router-dom";

test('Renders Correctly', () => {
    const store = createStore(rootReducer);
    const component = renderer.create(
        <Provider store={store}>
            <Router>
                <NavBar/>
            </Router>
        </Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});