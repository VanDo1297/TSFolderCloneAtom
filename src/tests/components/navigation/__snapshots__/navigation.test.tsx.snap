// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <span
    className="navbar-brand"
  >
    Bohe
  </span>
  <div
    className="mr-auto navbar-nav"
    onKeyDown={[Function]}
  >
    <a
      className="nav-link"
      data-rb-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Home
    </a>
    <a
      className="nav-link"
      data-rb-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Cards
    </a>
    <a
      className="nav-link"
      data-rb-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Charges
    </a>
    <a
      className="nav-link"
      data-rb-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Subscriptions
    </a>
    <a
      className="nav-link"
      data-rb-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Invoices
    </a>
    <a
      className="nav-link"
      data-rb-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Log Out
    </a>
  </div>
</nav>
`;
